src/scripts/migrate.py

```diff
@@ -0,0 +1,201 @@
+        # todo: I probably need only 1 of these
+        # rels_in: Dict = ontology.incoming_relationship_map(curie)
+        # rels_out: Dict = ontology.outgoing_relationship_map(curie)
+        pass
+
+
+    # TODO: How to get just the *direct* parents? look at relationships and see if rdfs:subClassOf?
+    #  ...Until above issues fixed, will need to get all relationships at beginning of run() func and use that
+    p = []
+    # for rel, parents in rels.items():
+    #     # print(f'  {rel} ! {ontology.get_label_by_curie(rel)}')
+    #     for parent in parents:
+    #         # print(f'    {parent} ! {ontology.get_label_by_curie(parent)}')
+    #         p.append(parent)
+    return p
+
+
+def _get_curies_all_owned_terms(
+    ontology: Union[SqlImplementation, ProntoImplementation], owned_prefix_map: Dict[str, str]
+) -> List[str]:
+    """Get all terms as CURIEs
+    ontology: Haven't decided yet which implementation I'll use. Would use superclass, but they use mult inheritance.
+    todo: owned_terms: if slow, can speed this up by grouping prefixes by splitting on : and filter out
+    """
+    # Get all terms: CURIES or URIs
+    terms = [x for x in ontology.entities()]
+
+    # Get CURIES
+    uri_terms = []
+    curie_terms_1 = []
+    for t in terms:
+        if t.startswith('http'):
+            uri_terms.append(t)
+        else:
+            curie_terms_1.append(t)
+
+    # - Convert from URI if needed
+    uri_converter = curies.Converter.from_prefix_map(owned_prefix_map)
+    curie_terms_2 = []
+    terms_with_no_known_prefix = []
+    for uri in uri_terms:
+        curie = uri_converter.compress(uri)
+        if curie:
+            curie_terms_2.append(curie)
+        else:
+            terms_with_no_known_prefix.append(uri)
+    curie_terms = curie_terms_1 + curie_terms_2
+
+    # Filter: Terms 'owned' by ontology
+    owned_terms = [x for x in curie_terms if any([x.startswith(y) for y in owned_prefix_map.keys()])]
+
+    return owned_terms
+
+
+def run(ontology_path: str, onto_config_path: str, sssom_map_path: str, min_id: str, mondo_terms_path: str, outpath: str) -> pd.DataFrame:
+    """Run slurp pipeline for given ontology"""
+    # Read inputs
+    # todo's: Ontology implementation selection
+    #  i. If trouble, can try `SparqlImplementation`, but ~6 min to load and queries slow (cuz rdflib)
+    #  ii. use Sql > Pronto when .entities() fixed: https://github.com/INCATools/ontology-access-kit/issues/235
+    with open(onto_config_path, 'r') as stream:
+        onto_config = yaml.safe_load(stream)
+    # ontology = SqlImplementation(OntologyResource(slug=ontology_path, local=True))
+    ontology = ProntoImplementation(OntologyResource(slug=ontology_path, local=True))
+    sssom_df = pd.read_csv(sssom_map_path, comment='#', sep='\t')
+    mondo_termlist_df = pd.read_csv(mondo_terms_path, comment='#', sep='\t')
+
+    # Get relationships
```

Comment:
FYI: This is how I determine the direct parents of a term, by first using OAK to get relationships and then building `    term_parent_map`.

Reply:
Actually, not doing it this way. Blah, all that work (not too much work) and I realized that actually, `[x for x in ProntoImplementation().relationships(subjects=[URI])]` will work in this case. I didn't notice before, because I was looking at the methods `ProntoImplementation().incoming_relationship_map` and `ProntoImplementation().outoing_relationship_map` which are broken right now. And I didn't notice `.relationships()`, which is defined in a parent class.